### PR TITLE
008.sh extended to check if UPF Scale Succeeded

### PR DESCRIPTION
**Pull request by rravindran123**

### CODE REVIEW

File: e2e/tests/008.sh [hunk 1]
```diff
@@ -28,8 +29,47 @@ source "${LIBDIR}/k8s.sh"
...
+    #Before scaling test get the running UPF POD ID
+    upf_pod_id=$(kubectl --kubeconfig $cluster_kubeconfig get pods -l app=free5gc-upf -l nf=upf -n free5gc-upf | grep upf | cut -d ' ' -f 1)
+
+    if [ -z "$upf_pod_id" ]; then
+    	echo "UPF PoD Not Found"
+    	exit 1
+    fi
+
+    #If the pod exists, Get the current CPU and Memory limit
+    current_cpu=$(kubectl --kubeconfig $cluster_kubeconfig describe pods $upf_pod_id -n free5gc-upf | awk '/cpu:/ {print $2}' | sed 's/m$//')
```
Comment (electrocucaracha) [inline e2e/tests/008.sh:65]: First, the [resource limits](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/) seem to be applied at container level and second, the label selection can result in multiple results.

So, in order to get the current CPU limit of all the containers for a set of pods with the same labels is:

```bash
kubectl get --kubeconfig $cluster_kubeconfig get pods -l app=free5gc-upf -l nf=upf -n free5gc-upf -o jsonpath='{range .items[*].spec.containers[*]}{.resources.limits.cpu}{"\n"}{end}'
```

But if you need to get the current CPU usage, the command is:

> Note: This requires Metrics API service installed on the server.

```bash
kubectl top pod --kubeconfig $cluster_kubeconfig get pods -l app=free5gc-upf -l nf=upf -n free5gc-upf
```

File: e2e/tests/008.sh [hunk 2]
```diff
@@ -28,8 +29,47 @@ source "${LIBDIR}/k8s.sh"
...
+
+    if [ "$after_scaling_cpu" -le  "$current_cpu" ] && [ "$after_scale_memory" -le  "$current_memory" ]; then
+        echo "UPF POD Scaling Failed"
+        exit 1
+    else
+        echo "UPF Pod Scaling Successful"
+    fi
+    
+    exit 0
+}
```
Comment (electrocucaracha) [inline e2e/tests/008.sh:48]: This function can be implemented in a more generic way:
```suggestion
function _check_scale {
    local metric=$1
    local previous=$2
    local current=$3
    
    echo "Comparing the new $metric after scaling"

    if [ "$previous" -le  "$current" ]; then
        echo "$metric scaling Failed"
        exit 1
    fi
    echo "$metric Pod Scaling Successful"
}
```

So the calling function will be:

```bash
_check_scale "CPU" $current_cpu $after_scaling_cpu
_check_scale "Memory" $current_memory $after_scaling_memory
```

**Review by electrocucaracha — CHANGES_REQUESTED** [review]

File: e2e/tests/008.sh (hunk 2 above)
Comment (rravindran123) [reply e2e/tests/008.sh:48]: Ok sure, will change it to this logic.

File: e2e/tests/008.sh (hunk 1 above)
Comment (rravindran123) [reply e2e/tests/008.sh:65]: Sure, will change it to filter based on  Json path

**Comment by johnbelamaric** [comment]
I am going to merge this, test it on a live system I have running now, then submit any needed tweaks in a separate PR.

/approve
/lgtm

**Pull request merged**